src/render/uniform_binding.js

```diff
@@ -133,13 +133,20 @@ class UniformMatrix4f extends Uniform<Float32Array> {
     }
 }
 
+class Uniform4fv extends Uniform<Float32Array> {
+    set(v: Float32Array): void {
+        this.gl.uniform4fv(this.location, v);
```

Comment:
Only relates to debug code so not a blocker, but have you evaluated the difference vs instancing with uniforms against a dynamic streaming vertex buffer thrown away and resized every frame? Stream buffer should only reallocate sparingly, I'm asking since uniform4v can be fairly slow and takes a high percentage of the frame time.

<img width="427" alt="Screen Shot 2020-03-18 at 12 12 47 PM" src="https://user-images.githubusercontent.com/7061573/76998189-e4ab1800-6911-11ea-864a-1dde04e1963f.png">

We should add a warning note here to prevent any usage of that in release builds for something else debug code path.

Reply:
This is a good catch. I don't think I have comparable performance numbers for different implementations. I'll have a look!

Reply:
Thanks for checking this! I think the new approach looks a bit simpler now as well.